gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/ReevaluateDagProc.java

```diff
@@ -60,21 +59,19 @@ protected Pair<Optional<Dag.DagNode<JobExecutionPlan>>, Optional<JobStatus>> ini
     Pair<Optional<Dag.DagNode<JobExecutionPlan>>, Optional<JobStatus>> dagNodeWithJobStatus =
         dagManagementStateStore.getDagNodeWithJobStatus(this.dagNodeId);
 
-    if (!dagNodeWithJobStatus.getLeft().isPresent() || !dagNodeWithJobStatus.getRight().isPresent()) {
+    if (!dagNodeWithJobStatus.getLeft().isPresent()) {
       // this is possible when MALA malfunctions and a duplicated reevaluate dag proc is launched for a dag node that is
       // already "reevaluated" and cleaned up.
       return ImmutablePair.of(Optional.empty(), Optional.empty());
     }
 
-    ExecutionStatus executionStatus = ExecutionStatus.valueOf(dagNodeWithJobStatus.getRight().get().getEventName());
-    if (!FlowStatusGenerator.FINISHED_STATUSES.contains(executionStatus.name())) {
-      log.warn("Job status for dagNode {} is {}. Re-evaluate dag action should have been created only for finished status - {}",
-          dagNodeId, executionStatus, FlowStatusGenerator.FINISHED_STATUSES);
-      // this may happen if adding job status in the store failed after adding a ReevaluateDagAction in KafkaJobStatusMonitor
-      throw new RuntimeException(String.format("Job status %s is not final for job %s", executionStatus, getDagId()));
+    if (dagNodeWithJobStatus.getRight().isPresent()) {
```

Comment:
somewhere let's add a comment describing circumstance of a dag node w/o job status.  where should that live?

Reply:
adding in next commit, search by "Usually reevaluate dag action"

Reply:
```// Usually reevaluate dag action is created by JobStatusMonitor when a finished job status is available,
      // but when reevaluate/resume/launch dag proc found multiple parallel jobs to run next, it creates reevaluate
      // dag actions for each of those parallel job and in this scenario there is no job status available.
      // If the job status is not present, this job was never launched, submit it now.```

Reply:
in multi job cases, at the dag start or at in the middle of the flow execution (say on the 4th job), whenever "next jobs to run" are more than one, we create reevaluate dag proc for **those next jobs**. next jobs status is not present. 
in resume case, status maybe actually present, if the dag ran in the earlier attempt.
did that answer your question?

Reply:
yes thank you.  do make sure the code comment reflects that nuance (rather than as written above w/ `when reevaluate/resume/launch dag proc...`